ledger-core/virtual/authentication/service.go

```diff
@@ -90,11 +90,11 @@ func (s service) getExpectedVE(ctx context.Context, subjectRef reference.Global,
 
 func (s service) IsMessageFromVirtualLegitimate(ctx context.Context, payloadObj interface{}, sender reference.Global, pr pulse.Range) (bool, error) {
 	verifyForPulse := pr.RightBoundData().PulseNumber
-	subjectRef, usePrev, ok := payload.GetSenderAuthenticationSubjectAndPulse(payloadObj)
+	subjectRef, mode, ok := payload.GetSenderAuthenticationSubjectAndPulse(payloadObj)
```

Comment:
I'd do this more like 
```
if !ok {
   return false, throw.New("Unexpected message type")
}

switch mode {
    case payload.UsePrevPulse:
    case payload.UseAnyPulse:
    default:
         panic(throw.IllegalValue())
}
```

But it's ok like this too

Reply:
will do in next pr